### PR TITLE
chore(sdk): Add SDK version header to allow for backwards compatibility code on the server

### PR DESCRIPTION
**Pull request by niklasnatter**

### CODE REVIEW

File: packages/sdk/src/swap/services/ApiService.ts [hunk 1]
```diff
@@ -24,12 +25,14 @@ export const getQuote: BackendQuery<
     }),
   };
 
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const queryParams = new URLSearchParams(params as Record<string, any>);
-
-  const url = new URL(`/quote?${queryParams.toString()}`, baseUrl).toString();
```
Comment (niklasnatter) [inline packages/sdk/src/swap/services/ApiService.ts:30]: found out that axios will do this for us while looking for the headers option

**Review by jerryafr — APPROVED** [review]

File: packages/sdk/src/swap/consts.ts [hunk 2]
```diff
@@ -1,4 +1,7 @@
 import { ChainflipNetworks } from '@/shared/enums';
+import { version } from '../../package.json';
+
+export const CF_SDK_VERSION_HEADERS = { 'Cf-Sdk-Version': version };
```
Comment (acdibble) [inline packages/sdk/src/swap/consts.ts:4]:
```suggestion
export const CF_SDK_VERSION_HEADERS = { 'X-Cf-Sdk-Version': version };
```

File: packages/sdk/src/swap/consts.ts [hunk 3]
```diff
@@ -1,4 +1,7 @@
 import { ChainflipNetworks } from '@/shared/enums';
+import { version } from '../../package.json';
```
Comment (acdibble) [inline packages/sdk/src/swap/consts.ts:2]: does this get inlined or does it have the entire package.json in the built output

File: packages/sdk/src/swap/services/ApiService.ts [hunk 4]
```diff
@@ -24,12 +25,12 @@ export const getQuote: BackendQuery<
...
-
-  const { data } = await axios.get<QuoteQueryResponse>(url, { signal });
+  const { data } = await axios.get<QuoteQueryResponse>('/quote', {
```
Comment (acdibble) [inline packages/sdk/src/swap/services/ApiService.ts:28]: we can use `axios.create` to create an instance that automatically sets the headers and then share it everywhere

File: packages/sdk/src/swap/consts.ts (hunk 3 above)
Comment (niklasnatter) [reply packages/sdk/src/swap/consts.ts:2]: i think its inlined - found this in the build: 
Screenshot 2024-07-23 at 10 03 11

File: packages/sdk/src/swap/services/ApiService.ts (hunk 4 above)
Comment (niklasnatter) [reply packages/sdk/src/swap/services/ApiService.ts:28]: unfortunately we cant pass the `baseUrl` via `axios.create` bc we pass it to the methods, so the instance would only contain the headers. no strong opinion about it

File: packages/sdk/src/swap/consts.ts (hunk 2 above)
Comment (niklasnatter) [reply packages/sdk/src/swap/consts.ts:4]: i asked chatgpt how to name the header yesterday lol, it told me that the `X-` prefix is not a recommendation anymore. but no strong opinion, im fine with adding it if you want

File: packages/sdk/src/swap/services/ApiService.ts (hunk 4 above)
Comment (acdibble) [reply packages/sdk/src/swap/services/ApiService.ts:28]: you can pass a base url: https://axios-http.com/docs/config_defaults

Comment (acdibble) [reply packages/sdk/src/swap/services/ApiService.ts:28]: i c

**Review by acdibble — APPROVED** [review]

**Pull request merged**